tm2/pkg/bft/state/store.go

```diff
@@ -172,6 +180,27 @@ func saveABCIResponses(db dbm.DB, height int64, abciResponses *ABCIResponses) {
 	db.SetSync(calcABCIResponsesKey(height), abciResponses.Bytes())
 }
 
+// LoadTxResult loads the tx result associated with the given
+// tx hash from the database, if any
+func LoadTxResult(db dbm.DB, txHash []byte) (*types.TxResult, error) {
```

Comment:
do we already store information per tx in the db?
if not perhaps we are introducing a large performance hit in indexing per tx (which arguably should live in the indexer).
i wonder if the best solution is to allow the tx indexer to plug in some RPC calls, so if you have an indexer, AND it supports fetching tx results, then we can call that pluggable RPC method into tm2 to be offerred.

Reply:
The indexer actually saves and serves this type of data on its own -- with much more bells and whistles (websocket support, filtering, graphQL...). The "bonus" features are something we've opted to not have in tm2, but on a separate service ([tx-indexer](https://github.com/gnolang/tx-indexer))

The node already serves data relating to the transactions:
- block info
- ABCI result info

This is the only method (index, really), that's missing for users to not run 2 separate services, but get data quickly from directly the node. It's just an index for transaction results that are saved anyways in the DB, with no fancy filtering included -- we wouldn't be expanding the RPC layer further, after this